File: Arrays/C++_Solutions/LC3300.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int minElement(vector<int>& nums) {
        int n = nums.size();
        int* ptr = nums.data();
        int* end = ptr + n;
        while (ptr<end)
        {
            int sum = 0;
            while (*ptr != 0)
            {
                sum += *ptr % 10;
                *ptr /= 10;
            }
            *ptr = sum;
            ptr++;
        }
        return *min_element(nums.begin(), nums.end());
    }
};

File: Arrays/Python_Solutions/LC3300.py
# LC3300: Minimum Element after replacement of digit sum : https://leetcode.com/problems/minimum-element-after-replacement-with-digit-sum/

from typing import List

class Solution:
    def minElement(self, nums: List[int]) -> int:        
        str_list = list(map(str, nums))
        k = 0
        for i in str_list:
            sum = 0
            for j in i:
                sum += int(j)
            nums[k] = sum
            k += 1
        nums.sort()
        return nums[0]

answer = Solution()
qes = [999,19,199]
print(answer.minElement(qes))